### ISSUE TITLE
timeout in some cases running SvanBoxel/organization-workflow@main step

### ISSUE THREAD

**Issue by gabriel-dantas98** [SvanBoxel/organization-workflows#32]
Tks @SvanBoxel for the project 😄 we are using internally to validate configuration files globally across the entire organization.
We currently have a large flow of pushes/​​commits with different frequencies between them.

In some executions, the **SvanBoxel/Organization-workflow@main** step runs for 6 minutes and we receive an "error" message:

![image](https://user-images.githubusercontent.com/20074620/114594519-b8241580-9c63-11eb-8eaa-3697461b4372.png)

```json
{
  "message":"The upstream server is timing out"
}

```
As we are using self-hosted runners this ends up creating a large queue of pending jobs.

We believe that the validity of the tokens may be related. _(maybe related to issue https://github.com/SvanBoxel/organization-workflows/issues/27)_

Perhaps the logs could be a bit more verbose too, currently, the only message we receive is: 
```
DEBUG (global-config-file/validator): GitHub request: POST https://api.github.com/repos/my-organization/sample-microservice/check-runs - 422
```

It will be a pleasure to help improve the project. 😉

**Comment by rkist** [comment]
We have the same problem here!

**Comment by SvanBoxel** [comment]
Hi @gabriel-dantas98 and @rkist, thanks for reporting this and oh-no-this-shouldn't-happen. 😮  I have a hunch what causes the timeout, could you share the timestamp of when you get the HTTP 422 so I can verify?

**Comment by gabriel-dantas98** [comment]
Of course 😉 this was a log that occurred today.

```log
2021-04-14T14:23:24.167876286Z github-organization-workflow DEBUG     (github): GitHub request: GET https://api.github.com/repos/my-organization/global-actions-workflows/contents/.github%2Fworkflows%2Fenforce-checklist-file.md - 200 
2021-04-14T14:23:24.325641237Z github-organization-workflow DEBUG     (checklist-as-code/validator): GitHub request: POST https://api.github.com/repos/my-organization/static-frontend-repo/check-runs - 422
2021-04-14T14:23:24.325674011Z github-organization-workflow     head_sha: "0000000000000000000000000000000000000000"
2021-04-14T14:23:24.325678477Z github-organization-workflow     details_url: "https://github.com/my-organization/static-frontend-repo/actions/runs/748634312"
2021-04-14T14:23:24.325682166Z github-organization-workflow     status: "in_progress"
2021-04-14T14:23:24.325685325Z github-organization-workflow     output: {
2021-04-14T14:23:24.325688169Z github-organization-workflow       "title": "checklist-as-code/validator",
2021-04-14T14:23:24.325691441Z github-organization-workflow       "summary": "# Checklist as Code global check action\n\nThis workflow checks whether you've used checklist in your repository properly.\nYou can see more information about checklist [here]()"
2021-04-14T14:23:24.325696973Z github-organization-workflow     }
2021-04-14T14:23:24.325846970Z github-organization-workflow (node:53) UnhandledPromiseRejectionWarning: HttpError: No commit found for SHA: 0000000000000000000000000000000000000000 
2021-04-14T14:23:24.325866440Z github-organization-workflow     at /usr/src/app/node_modules/@octokit/request/dist-node/index.js:66:23
2021-04-14T14:23:24.325870415Z github-organization-workflow     at runMicrotasks (<anonymous>)
2021-04-14T14:23:24.325873787Z github-organization-workflow     at processTicksAndRejections (internal/process/task_queues.js:97:5)
2021-04-14T14:23:24.325876690Z github-organization-workflow     at async sendRequestWithRetries (/usr/src/app/node_modules/@octokit/auth-app/dist-node/index.js:440:12)
2021-04-14T14:23:24.325879622Z github-organization-workflow     at async Job.doExecute (/usr/src/app/node_modules/bottleneck/light.js:405:18)
2021-04-14T14:23:24.325886908Z github-organization-workflow (node:53) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 78)
2021-04-14T14:23:25.212856246Z github-organization-workflow INFO     (http): POST / 200 - 0ms
```

I also noticed that the log message says that the head_sha received is `0000000000000000000000000000000000000000`

**Comment by gabriel-dantas98** [comment]
Good news @SvanBoxel 😄 
I spent the day investigating the behavior of events that are sent by the `repository_dispatcher`.
I noticed that all 422 errors happened when `client_payload.sha` contained several zero values.

I did some tests here and I realized that this happens the moment after the pull request is approved and merged, the moment the branch is deleted from the repository.
![image](https://user-images.githubusercontent.com/20074620/114780087-4d4f0900-9d4d-11eb-99ae-94f704d0f1da.png)

Our solution was to skip the execution when this configuration happens, using an if at the job level.
```yaml
jobs:
  register:
    runs-on: self-runner
    if: github.event.client_payload.sha != '0000000000000000000000000000000000000000'
    steps:
    ......
```
I imagine that maybe the idea is to deal with this before sending the event or maybe at the moment of the execution of the step.
